.github/workflows/_buildpacks-release.yml

```diff
@@ -60,12 +60,18 @@ env:
 jobs:
   compile:
     name: Compile Buildpacks
-    runs-on: ubuntu-24.04
+    runs-on: ${{ inputs.ip_allowlisted_runner }}
     outputs:
       buildpacks: ${{ steps.generate-buildpack-matrix.outputs.buildpacks }}
       version: ${{ steps.generate-buildpack-matrix.outputs.version }}
       changelog: ${{ steps.generate-changelog.outputs.changelog }}
     steps:
+      - name: Get token for GH application (Linguist)
```

Comment:
We only need to fetch the token if we're doing some type of writable action against the repo, right? I don't think we need this for the checkout step. It should work for private repos as long as it's using an allowlisted runner.

Reply:
If checkout for read-only works without the GH app token, then I agree this is unnecessary.